js/tsconfig.json

```diff
@@ -3,9 +3,9 @@
   "display": "Node 14",
 
   "compilerOptions": {
-    "lib": ["es2020"],
+    "lib": ["esnext"],
```

Comment:
I imagine this is fine for now; the plan I imagine is:
```mermaid
graph LR
    A[.ts] -->|ts AST via tsc| B((ts AST))
    B -->|map to LST| C((LST))
    C -->|.ts file via printer| D[.ts file]
```

Reply:
Yes, the supported extensions are set to:

- .js
- .jsx
- .mjs
- .cjs
- .ts
- .tsx
- .mts
- .cts

There are a few configurations I am unsure of:

1. Should we set the ScriptTarget based on the project? ES2000, ES2020, ESNEXT, etc.
2. Is it safe to set the ScriptKind to the TS equivalent of a JS file (JS -> TS, JSX -> TSX), or should we let the extension set the behavior in the compiler?

Reply:
I don't know what the ScriptTarget settings affect in the compiler. I changed it so that the Scanner would be set to the same version of the `lib` and `target` in the compiler options.